festim/hydrogen_transport_problem.py

```diff
@@ -711,3 +713,450 @@ def post_processing(self):
+            # and get the two cells that are connected to it
+            # and get the material properties of these cells
+            first_facet_interface = mt.find(interface.id)[0]
+            c_plus, c_minus = (
+                f_to_c.links(first_facet_interface)[0],
+                f_to_c.links(first_facet_interface)[1],
+            )
+            id_minus, id_plus = ct.values[c_minus], ct.values[c_plus]
+
+            for subdomain in interface.subdomains:
+                if subdomain.id == id_plus:
+                    subdomain_1 = subdomain
+                if subdomain.id == id_minus:
+                    subdomain_2 = subdomain
+
+            all_mobile_species = [spe for spe in self.species if spe.mobile]
+            if len(all_mobile_species) > 1:
+                raise NotImplementedError("Multiple mobile species not implemented")
+            H = all_mobile_species[0]
+
+            v_b = H.subdomain_to_test_function[subdomain_1](b_res)
+            v_t = H.subdomain_to_test_function[subdomain_2](t_res)
+
+            u_b = H.subdomain_to_solution[subdomain_1](b_res)
+            u_t = H.subdomain_to_solution[subdomain_2](t_res)
+
+            K_b = subdomain_1.material.get_solubility_coefficient(
+                self.mesh.mesh, self.temperature_fenics(b_res), H
+            )
+            K_t = subdomain_2.material.get_solubility_coefficient(
+                self.mesh.mesh, self.temperature_fenics(t_res), H
+            )
+
+            F_0 = -0.5 * mixed_term((u_b + u_t), v_b, n_b) * dInterface(
+                interface.id
+            ) - 0.5 * mixed_term(v_b, (u_b / K_b - u_t / K_t), n_b) * dInterface(
+                interface.id
+            )
+
+            F_1 = +0.5 * mixed_term((u_b + u_t), v_t, n_b) * dInterface(
+                interface.id
+            ) - 0.5 * mixed_term(v_t, (u_b / K_b - u_t / K_t), n_b) * dInterface(
+                interface.id
+            )
+            F_0 += (
+                2
+                * gamma
+                / (h_b + h_t)
+                * (u_b / K_b - u_t / K_t)
+                * v_b
+                * dInterface(interface.id)
+            )
+            F_1 += (
+                -2
+                * gamma
+                / (h_b + h_t)
+                * (u_b / K_b - u_t / K_t)
+                * v_t
+                * dInterface(interface.id)
+            )
+
+            subdomain_1.F += F_0
+            subdomain_2.F += F_1
+
+        J = []
+        forms = []
+        for subdomain1 in self.volume_subdomains:
+            jac = []
+            form = subdomain1.F
+            for subdomain2 in self.volume_subdomains:
+                jac.append(
+                    dolfinx.fem.form(
+                        ufl.derivative(form, subdomain2.u), entity_maps=entity_maps
+                    )
+                )
+            J.append(jac)
+            forms.append(dolfinx.fem.form(subdomain1.F, entity_maps=entity_maps))
```

Comment:
Thanks to https://github.com/FEniCS/dolfinx/pull/3450 we should be able to simplify this bit

Reply:
Done